platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/ConnectionServer.java

```diff
@@ -48,32 +52,32 @@ public class ConnectionServer implements InterruptableRunnable {
     /** a thread pool used to handle incoming connections */
     private final ExecutorService incomingConnPool;
 
+    private final List<PeerInfo> peerInfoList;
     /**
-     *  @param threadManager
-     *  		responsible for managing thread lifecycles
-     * @param port
-     * 		the port ot use
-     * @param socketFactory
-     * 		responsible for creating new sockets
-     * @param newConnectionHandler
-     * 		handles a new connection after it has been created
+     * @param threadManager        responsible for managing thread lifecycles
+     * @param port                 the port ot use
+     * @param socketFactory        responsible for creating new sockets
+     * @param newConnectionHandler handles a new connection after it has been created
+     * @param peerInfoList         List of all peers
      */
     public ConnectionServer(
             final ThreadManager threadManager,
             final int port,
             final SocketFactory socketFactory,
-            final Consumer<Socket> newConnectionHandler) {
+            final Consumer<Socket> newConnectionHandler,
+            final List<PeerInfo> peerInfoList) {
```

Comment:
can this be annotated `@NonNull`?

Reply:
It can but I opted for consistency - either we annotate all as @NonNull or leave as-is. I opted for as-is.

Reply:
IMO there is no problem with inconsistency in this regard.

We want to move toward every parameter having an annotation. If annotating all existing parameters is outside the scope of a PR, at least annotating newly introduced parameters moves us closer to our final goal, rather than building more tech debt to be paid later

Reply:
That's acceptable. Added the annotation to just that one new parameter.